The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/services/GeocodingService.java

```diff
@@ -0,0 +1,80 @@
+package com.example.The_Coding_Sharks_demo.services;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import com.example.The_Coding_Sharks_demo.models.DTO.GeocodingResultDTO;
+import com.example.The_Coding_Sharks_demo.models.Destination;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import org.springframework.stereotype.Service;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Service
+public class GeocodingService {
+    private final HttpClient httpClient;
+    private final ObjectMapper objectMapper;
+
+    public GeocodingService() {
+        this.httpClient = HttpClient.newHttpClient();
+        this.objectMapper = new ObjectMapper();
+    }
+
+    public Destination geocodeDestination(String destinationName) throws Exception {
+        String apiKey = "3843f958ea2f4ba0beca427b6f821857";
```

Comment:
We may want to revisit this later if we have time - this works fine now for our purposes but I think best practice is to not hardcode the API key.

Reply:
ok. I didn't realize that, but it totally makes sense. I will create another issue for this.